File: domain/src/main/java/com/ortin/gesturetranslator/domain/usecases/RecognizeImageUseCase.java
package com.ortin.gesturetranslator.domain.usecases;

import com.ortin.gesturetranslator.domain.listeners.RecognizeImageListener;
import com.ortin.gesturetranslator.domain.models.Image;
import com.ortin.gesturetranslator.domain.repository.RecognizeImageRepository;

public class RecognizeImageUseCase {
    private RecognizeImageRepository recognizeImageRepository;

    public RecognizeImageUseCase(RecognizeImageRepository recognizeImageRepository) {
        this.recognizeImageRepository = recognizeImageRepository;
    }

    public void execute(Image image){
        recognizeImageRepository.recogniseImage(image);
    }

    public void setOnRecogniseListener(RecognizeImageListener recognizeImageListener){
        recognizeImageRepository.setRecogniseListener(recognizeImageListener);
    }
}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/usecases/RecognizeImageUseCase.kt
package com.ortin.gesturetranslator.domain.usecases

import com.ortin.gesturetranslator.domain.listeners.RecognizeImageListener
import com.ortin.gesturetranslator.domain.models.Image
import com.ortin.gesturetranslator.domain.repository.RecognizeImageRepository
import javax.inject.Inject

class RecognizeImageUseCase @Inject constructor(private val recognizeImageRepository: RecognizeImageRepository) {
    fun execute(image: Image) {
        recognizeImageRepository.recogniseImage(image)
    }

    fun setOnRecogniseListener(recognizeImageListener: RecognizeImageListener) {
        recognizeImageRepository.setRecogniseListener(recognizeImageListener)
    }
}
